File: task3/GO/main.go
package main

import (
	"fmt"
	"math"
)

// getGcd находит наибольший общий делитель (алгоритм Евклида)
func getGcd(a, b int64) int64 {
	for b != 0 {
		a %= b
		a, b = b, a
	}
	return a
}

func main() {
	var a, b int
	// Считываем входные данные
	_, err := fmt.Scan(&a, &b)
	if err != nil {
		return
	}

	// Условие расходимости: если основание b <= 1, сумма стремится к бесконечности
	if b <= 1 {
		fmt.Println("infinity")
		return
	}

	// Накапливаем сумму ряда
	var sum float64
	for n := 1.0; n <= 500; n++ {
		term := math.Pow(n, float64(a)) / math.Pow(float64(b), n)
		if term < 1e-19 {
			break
		}
		sum += term
	}

	// Поиск подходящей дроби методом перебора знаменателя
	var bestNum, bestDen int64
	minError := 1e-15

	for den := int64(1); den <= 10000; den++ {
		// round в Go возвращает float64, приводим к int64
		num := int64(math.Round(sum * float64(den)))
		if math.Abs(sum-float64(num)/float64(den)) < minError {
			bestNum = num
			bestDen = den
			break
		}
	}

	// Сокращаем дробь через НОД и выводим результат
	common := getGcd(bestNum, bestDen)
	if bestDen != 0 {
		fmt.Printf("%d/%d\n", bestNum/common, bestDen/common)
	}
}

File: task3/CPP/main.cpp
#include <iostream>
#include <cmath>

using namespace std;

// Функция для сокращения дроби (Нахождение наибольшего общего делителя)
long long getGcd(long long a, long long b) {
    while (b) {
        a %= b;
        swap(a, b);
    }
    return a;
}

int main() {
    int a, b;
    if (!(cin >> a >> b)) return 0;

    // Условие расходимости
    if (b <= 1) {
        cout << "infinity" << endl;
        return 0;
    }

    // Будем накапливать сумму в double для точности
    long double sum = 0;
    for (int n = 1; n <= 500; ++n) {
        long double term = pow((long double)n, a) / pow((long double)b, n);
        if (term < 1e-19) break; // Ряд сходится быстро, этого достаточно
        sum += term;
    }

    // Теперь превратим десятичное число обратно в дробь
    // Используем упрощенный метод подбора знаменателя
    long long bestNum = 0, bestDen = 1;
    long double minerror = 1e-15;

    for (long long den = 1; den <= 10000; ++den) {
        long long num = (long long)round(sum * den);
        if (abs(sum - (long double)num / den) < minerror) {
            bestNum = num;
            bestDen = den;
            break;
        }
    }

    // Сокращаем результат перед выводом
    long long common = getGcd(bestNum, bestDen);
    cout << bestNum / common << "/" << bestDen / common << endl;

    return 0;
}